src/Kentico.Search/SearchService.cs

```diff
@@ -53,13 +54,13 @@ public SearchService(string[] searchIndexNames, string cultureName, string siteN
 
 
         /// <summary>
-        /// Performs full-text search.
+        /// Performs paged full-text search.
         /// </summary>
-        /// <param name="query">Text to search</param>
-        /// <param name="page">Zero-based page index</param>
-        /// <param name="pageSize">Specifies the number of results per page</param>
-        /// <param name="numberOfResults">Total number of search results</param>
-        /// <returns>Search results or null if <paramref name="query"/> is empty.</returns>
+        /// <param name="query">Text to search.</param>
+        /// <param name="page">Zero-based page index.</param>
+        /// <param name="pageSize">Specifies the number of results per page. Page size must be a positive number.</param>
+        /// <param name="numberOfResults">Total number of search results for the given query which can be retrieved from the search index.</param>
+        /// <returns>Search results for the specified page or null if <paramref name="query"/> is empty.</returns>
         public virtual IEnumerable<SearchResultItem> Search(string query, int page, int pageSize, out int numberOfResults)
```

Comment:
Parameter page should be name pageIndex to match SearchInternal method. It's also more clear what is this parameter about. 
(If it's not a breaking change).